src/server/mod.rs

```diff
@@ -96,7 +127,14 @@ impl Server {
                 port,
             };
 
-            build_app(state)
+            let dfstate = ServerState {
```

Comment:
I really hope I'm being dumb here and there's a better way to send the ServerState to the second App instance. I tried a few things, but they got ugly really fast. Setting up the pool and Arc clones seems wasteful, since the app never uses them.

Reply:
ServerState could support a clone with some effort. It would have to manually call Db's box_clone.

The dockerflow APIs don't need ServerState currently, so you could not pass it at all (apps can have [different state via .boxed()](https://actix.rs/docs/application/#combining-applications-with-different-state)) but at some point `__heartbeat__` should be doing a database check anyway..

Reply:
actix-web 1.0 might change this anyway? It doesn't have an App::with_state any more (unfortunately the user guide hasn't been updated yet https://github.com/actix/actix-website/tree/master/content/docs)

Reply:
Ah, yeah, things might change a lot with the actix update. This PR was mostly just trying to get ahead of that update so things weren't joined and the PR was even uglier. 

I agree that `__heartbeat__` should do a bit more than just return Ok, but again, that could be added later.